File: publish_tool/lib/dto/server_os_info_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'server_os_info_dto.g.dart';

@JsonSerializable()
class ServerOsInfoDto {
  String os;
  String platform;
  String goARCH;
  String version;
  int numCPU;
  String cpuName;
  double cpuMhz;
  double diskUsed;
  double diskFree;
  double diskTotal;
  double diskUsedPercent;

  ServerOsInfoDto(
    this.os,
    this.platform,
    this.goARCH,
    this.version,
    this.numCPU,
    this.cpuName,
    this.cpuMhz,
    this.diskUsed,
    this.diskFree,
    this.diskTotal,
    this.diskUsedPercent,
  );

  factory ServerOsInfoDto.fromJson(Map<String, dynamic> json) =>
      _$ServerOsInfoDtoFromJson(json);
  Map<String, dynamic> toJson() => _$ServerOsInfoDtoToJson(this);
}

File: publish_tool_avalonia/src/PublishTool/Models/ServerOsInfoDto.cs
using Newtonsoft.Json;

namespace PublishTool.Models;

public class ServerOsInfoDto
{
    [JsonProperty("os")]
    public string? Os { get; set; }

    [JsonProperty("platform")]
    public string? Platform { get; set; }

    [JsonProperty("goARCH")]
    public string? GoArch { get; set; }

    [JsonProperty("version")]
    public string? Version { get; set; }

    [JsonProperty("numCPU")]
    public int NumCpu { get; set; }

    [JsonProperty("cpuName")]
    public string? CpuName { get; set; }

    [JsonProperty("cpuMhz")]
    public double CpuMhz { get; set; }

    [JsonProperty("diskUsed")]
    public double DiskUsed { get; set; }

    [JsonProperty("diskFree")]
    public double DiskFree { get; set; }

    [JsonProperty("diskTotal")]
    public double DiskTotal { get; set; }

    [JsonProperty("diskUsedPercent")]
    public double DiskUsedPercent { get; set; }
}
